falcon/media/multipart.py

```diff
@@ -616,6 +627,7 @@ def __init__(self) -> None:
         self.default_charset = 'utf-8'
         self.max_body_part_buffer_size = 1024 * 1024
         self.max_body_part_count = 64
+        self.max_secure_filename_length = 96
```

Comment:
We cannot afford to make this breaking change in a SemVer-minor version.
The default needs to be `None`, with a note that we are likely to change this to finite value like 96 in Falcon 5.0.